src/ceph-volume/ceph_volume/util/device.py

```diff
@@ -386,16 +386,20 @@ def used_by_ceph(self):
 
 
     def _check_generic_reject_reasons(self):
-        reasons = [
-            ('removable', 1, 'removable'),
-            ('ro', 1, 'read-only'),
-            ('locked', 1, 'locked'),
-        ]
-        rejected = [reason for (k, v, reason) in reasons if
-                    self.sys_api.get(k, '') == v]
-        # reject disks small than 5GB
-        if int(self.sys_api.get('size', 0)) < 5368709120:
-            rejected.append('Insufficient space (<5GB)')
+        rejected = []
+        if self.sys_api:
+            reasons = [
+                ('removable', 1, 'removable'),
+                ('ro', 1, 'read-only'),
+                ('locked', 1, 'locked'),
+            ]
+            rejected = [reason for (k, v, reason) in reasons if
+                        self.sys_api.get(k, '') == v]
+            # reject disks smaller than 5GB
+            if int(self.sys_api.get('size', 0)) < 5368709120:
+                rejected.append('Insufficient space (<5GB)')
+        else:
+            rejected.append("Device type is not acceptable. It should be disk or partition")
```

Comment:
I think it would be better to add an explicit property to `Device` like `is_valid_device` or so.

This reads as "if `sys_api` is not present, its not a valid device". Its less then obvious why that is I find. Being explicit what constitutes a valid device would be better imho. Or `is_partition or is_device` would also work.

Reply:
Fixed, thanks. I choose to add is_valid_evice() because it's more descriptive.